src/components/InterestForm/index.tsx

```diff
@@ -1,73 +1,101 @@
+  start_date?: Date;
+  needs_interpreter?: boolean;
+  interest_reason: string;
+}
 
-export default function InterestForm({ caseData }: { caseData: CaseListing }) {
+export default function InterestForm({
+  listingData,
+  interpretation = false,
+}: {
+  listingData: Listing;
+  interpretation?: boolean;
+}) {
   const auth = useAuth();
   const [reason, setReason] = useState<string>('');
-  const [rolesInterested, setRolesInterested] = useState<string>('');
+  const [needsInterpreter, setNeedsInterpreter] = useState<string>('');
   const [startDate, setStartDate] = useState<string>('');
   const [submitted, setSubmitted] = useState(false);
   const [missingInfo, setMissingInfo] = useState(false);
 
   useEffect(() => {
     // Reset form fields when caseData changes
     setReason('');
+    setNeedsInterpreter('');
     setStartDate('');
-    setRolesInterested('');
     setSubmitted(false);
     setMissingInfo(false);
-  }, [caseData]);
+  }, [listingData]);
 
   const handleInsert = async () => {
-    if (startDate === '' || rolesInterested === '') {
+    // Error handling, check if required fields are unfilled.
+    if (
+      // CASE listing requires: startDate and needsInterpreter
+      (listingData.listing_type === 'CASE' &&
+        !interpretation &&
+        (startDate === '' || needsInterpreter === '')) ||
+      // CASE_INT and INT listings require: startDate
+      (((listingData.listing_type === 'CASE' && interpretation) ||
+        listingData.listing_type === 'INT') &&
+        startDate === '') ||
+      // LCA and DOC listings require: reason
+      ((listingData.listing_type === 'LCA' ||
+        listingData.listing_type === 'DOC') &&
+        reason === '')
+    ) {
       setMissingInfo(true);
       return;
     }
-    if (isValidDate(startDate)) {
+    const responses: Responses = { interest_reason: '' };
+    if (
+      isValidDate(startDate) ||
+      listingData.listing_type === 'DOC' ||
+      listingData.listing_type === 'LCA'
+    ) {
       if (auth && auth.userId) {
+        if (
+          listingData.listing_type === 'CASE' ||
+          listingData.listing_type === 'INT'
+        ) {
+          responses.start_date = new Date(startDate);
+          if (listingData.listing_type === 'CASE' && !interpretation) {
+            responses.needs_interpreter = needsInterpreter === 'Yes';
+          }
+        }
+        responses.interest_reason = reason;
         const newInterest: Interest = {
-          listing_id: caseData.id,
-          listing_type: 'CASE',
+          listing_id: listingData.id,
+          listing_type:
+            listingData.listing_type === 'CASE' && interpretation
+              ? 'CASE_INT'
+              : listingData.listing_type,
           user_id: auth.userId,
           form_response: {
-            start_date: new Date(startDate),
-            interest_reason: reason,
-            // TODO: add needs_interpreter field
+            ...responses,
           },
```

Comment:
instead of spreading `responses` as you've done here, we can just do `form_responses: responses`

Reply:
(mansplaining alert)
spreading technically creates a new object and copies the fields of `reponses` over, which is really only useful if we're adding extra fields besides those already in `responses`.
but since `responses` is all we're passing in, there's no need to use more compute resources to copy the object

Reply:
gotcha, done !